nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/functions/PropertyUtils.java

```diff
@@ -202,13 +205,33 @@ public static BlobTextFromDocument docSerialize(DocumentModel doc, Set<String> p
         });
 
         if (log.isDebugEnabled() && properties.size() + blobTextFromDoc.getBlobs().size() != propertiesList.size()) {
-            log.debug(String.format("Document %s one of the following properties is null. %s",
-                                    doc.getId(), propertiesList));
+            log.debug(String.format("Document %s one of the following properties is null. %s", doc.getId(),
+                    propertiesList));
         }
 
         return blobTextFromDoc;
     }
 
+    /**
+     * Get the best conversion of the original blob if exist. If not return the original blob.
+     */
+    private static ManagedBlob getPictureConversion(DocumentModel doc, ManagedBlob originalContent) {
```

Comment:
Do we have any other ticket to check that the conversion is connected to the final image, and not a previous one ?

Reply:
There is a ticket for that to do later, but apparently, the code i set shoudl be already safe

Reply:
https://jira.nuxeo.com/browse/NXP-28435